flink-yarn/src/main/java/org/apache/flink/yarn/entrypoint/YarnEntrypointUtils.java

```diff
@@ -98,9 +98,9 @@ public static Configuration loadConfiguration(String workingDirectory, Map<Strin
 			configuration.setInteger(WebOptions.PORT, 0);
 		}
 
-		if (configuration.getInteger(RestOptions.PORT) >= 0) {
+		if (!configuration.getString(RestOptions.BIND_PORT).equals("0")) {
```

Comment:
"00" is equivalent but would fail this check.

Reply:
This whole check seems unnecessary; if it is non-zero it is set to 0, otherwise it was set to zero to being with. We can just always set this to 0, without actually changing any behavior.